app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt

```diff
@@ -63,12 +63,7 @@ class EditPatientEffectHandler @AssistedInject constructor(
       ObservableTransformer<FetchBangladeshNationalIdEffect, EditPatientEvent> { effectStream ->
         effectStream
             .flatMap { patientRepository.bangladeshNationalIdForPatient(it.patient.uuid).subscribeOn(scheduler) }
-            .map {
-              when (it) {
-                is Just -> it.value.identifier.value
-                None -> ""
-              }
-            }
+            .filterAndUnwrapJust()
```

Comment:
Yes. This is what I was talking about :)

Reply:
Cool :)